pkg/storage/types/dv.go

```diff
@@ -84,9 +84,14 @@ func GetResolvedCloneSource(ctx context.Context, client kubecli.KubevirtClient,
 			return nil, err
 		}
 
+		resolvedSource := ds.Spec.Source.DeepCopy()
+		if ds.Spec.Source.DataSource != nil {
+			resolvedSource = ds.Status.Source.DeepCopy()
+		}
```

Comment:
https://github.com/kubevirt/kubevirt/blob/fa01e3f6210d394a27c341be5365d864fd2397c6/pkg/storage/types/dv.go#L72 I planned to move it to the API package before, which could have bailed us out here and allowed external projects to embed those too